inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/sink/mysql/MySQLSinkDTO.java

```diff
@@ -222,18 +231,40 @@ public static String filterSensitive(String url) {
         if (StringUtils.isBlank(url)) {
             return url;
         }
+
         try {
             String resultUrl = url;
             while (resultUrl.contains(InlongConstants.PERCENT)) {
                 resultUrl = URLDecoder.decode(resultUrl, "UTF-8");
             }
             resultUrl = resultUrl.replaceAll(InlongConstants.BLANK, "");
-            for (String sensitiveParam : SENSITIVE_PARAM_MAP.keySet()) {
-                if (StringUtils.containsIgnoreCase(resultUrl, sensitiveParam)) {
-                    resultUrl = StringUtils.replaceIgnoreCase(resultUrl, sensitiveParam,
-                            SENSITIVE_PARAM_MAP.get(sensitiveParam));
+
+            if (resultUrl.contains(InlongConstants.QUESTION_MARK)) {
+                StringBuilder builder = new StringBuilder();
+                builder.append(StringUtils.substringBefore(resultUrl, InlongConstants.QUESTION_MARK));
+                builder.append(InlongConstants.QUESTION_MARK);
+
+                LinkedHashMap<String, String> queryMap = new LinkedHashMap<>();
+                String queryString = StringUtils.substringAfter(resultUrl, InlongConstants.QUESTION_MARK);
+                for (String param : queryString.split("&")) {
+                    queryMap.put(StringUtils.substringBefore(param, "="),
+                            StringUtils.substringAfter(param, "="));
                 }
+
+                queryMap.entrySet().removeIf(entry -> SENSITIVE_REMOVE_PARAM_MAP.contains(entry.getKey()));
+                queryMap.entrySet().forEach(entry -> {
+                    if (SENSITIVE_REPLACE_PARAM_MAP.containsKey(entry.getKey())) {
+                        entry.setValue(SENSITIVE_REPLACE_PARAM_MAP.get(entry.getKey()));
+                    }
+                });
+
+                String params = StringUtils.join(
```

Comment:
Only need to traverse once instead of repeatedly.

Reply:
OK, fix it.